### ISSUE TITLE
StructPtr cannot be used to represent data passed from client to XCB

### ISSUE THREAD

**Issue by darkstalker** [rtbo/rust-xcb#7]
Can't really do much with this library since i don't know where to go for reference. Things don't exactly match the C library (i'm following this [1]). For example, if I try to use `xcb::poly_point`, when passing the points I get an error like:
```
src/main.rs:45:21: 45:29 error: mismatched types:
 expected `xcb::base::StructPtr<'_, xcb::ffi::xproto::xcb_point_t>`,
    found `(_, _)`
(expected struct `xcb::base::StructPtr`,
    found tuple) [E0308]
src/main.rs:45                     (10, 10), (10, 20), (20, 10)
                                   ^~~~~~~~
```
And don't know how to generate a `xcb::base::StructPtr<'_, xcb::ffi::xproto::xcb_point_t>` .. grepping through the sources doesn't help, since things are auto-generated. Don't know if I should use a helper function or something.

[1] http://www.x.org/releases/X11R7.6/doc/libxcb/tutorial/index.html

**Comment by rtbo** [comment]
@darkstalker
the problem is that the bindings are wrong in this case. `StructPtr` is made to wrap a pointer to a struct that you would get from XCB. 
In this case, the definition of `Point` should be: `type Point = xcb_point_t` such as you can build an array of points and pass a slice of it to xcb::poly_point.
I need to find a way for discerning value types from reference types. I know at least xcb::Setup _must_ use StructPtr.

Thanks for reporting this. My personal use of xcb only turns around window management only and I did not encounter the problem

Regarding documentation: most of XCB's documentation is generated from the XML. So beside an additional example, the documentation I can build will not be much better than XCB's documentation.

**Comment by rtbo** [comment]
Solved by having distinction between POD types wrapped by value with helper constructor and other types (wrapped by `StructPtr`) that must be obtained from XCB. 
I just hope that the distinction is made on good basis. This will surface again eventually.

@darkstalker fix is availble as `v0.6.0` on [crates.io](https://crates.io/crates/xcb). See also [drawing.rs](https://github.com/rtbo/rust-xcb/blob/master/examples/drawing.rs) example.

@laumann it seems after all that `StructPtr` cannot be used everywhere like we assumed.

**Comment by ibabushkin** [comment]
I *think* some types still require some changes. For example `xkb::SetMapValues` and `xkb::SelectEventsDetails` both are afaict expected to be constructed by the user and passed to XCB then. Hence, the following definition:

```rust
pub type SelectEventsDetails<'a> = base::StructPtr<'a, xcb_xkb_select_events_details_t>;
```

should be replaced by a POD type.

I'll draft up a PR with the necessary changes and see how this goes.
